tests/firewall/tests.c

```diff
@@ -39,6 +39,42 @@
 #include "utils/mac.h"
 #include "utils/bitmask.h"
 
+static void test_firewall_rules(void)
+{
+	struct pg_error *error = NULL;
+	struct pg_brick *fw = pg_firewall_new("fw", 1, 1, PG_NONE, &error);
+	const char *r[] = {
+		"src host 10.0.0.1",
+		"udp",
+		"tcp",
+		"icmp",
+		"icmp or udp or tcp",
+		"udp dst portrange 4000-6000",
+		"tcp dst portrange 0-65535",
+		"tcp dst port 42",
+		"src host 10.0.0.1 and (icmp or udp or tcp)",
+		"((src net 0.0.0.0/0 and udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 && udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 or udp dst portrange 4000-6000))",
+		"((src net 0.0.0.0/0 || udp dst portrange 4000-6000))",
+		NULL
+	};
+
+	for (int i = 0; r[i]; i++) {
+		if (pg_firewall_rule_add(fw, r[i], WEST_SIDE, 0, &error) < 0) {
```

Comment:
all you if is useless.
you can do: 
`g_assert(!pg_firewall_rule_add(fw, r[i], WEST_SIDE, 0, &error))`
`g_assert(!error);`

or `if (error) pg_error_print(error);` if you really want to print the error in case of failure